### PR TITLE
Add TabbedHeader boxel-ui component

### PR DESCRIPTION
**Pull request by burieberry**
Boxel UI TabbedHeader component preview link:
https://tabbed-header-component.boxel-ui-preview.stack.cards/?s=Components&ss=%3CTabbedHeader%3E#favorites

Used it in app-card and ai app generator card in drafts realm

### CODE REVIEW

File: packages/boxel-ui/addon/src/components/tabbed-header/usage.gts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+          @tabs={{this.tabs}}
+          @onSetActiveTab={{fn this.onSetActiveTab}}
+          @activeTabIndex={{this.activeTabIndex}}
+          @iconComponent={{this.iconComponent}}
+          @iconURL={{this.iconURL}}
+          @headerBackgroundColor={{this.headerColor}}
+          @iconBackgroundColor={{this.iconBackgroundColor}}
+          @iconBorderColor={{this.iconBorderColor}}
+          @iconCoversAllAvailableSpace={{this.iconCoversAllAvailableSpace}}
+        />
```
Comment (backspace) [inline packages/boxel-ui/addon/src/components/tabbed-header/usage.gts]: With all these icon-related properties, might it be cleaner to use a named `:icon` block or the like to provide customisation?

**Review by backspace — APPROVED** [review]
This looks great 🤩

I’ve seen this subtle but fun animation in this kind of interface, I wonder if it’s easy to implement

cscreencast 2024-08-02 10-35-10 2024-08-02 10_38_00

**Review by habdelra — APPROVED** [review]

File: packages/boxel-ui/addon/src/components/tabbed-header/usage.gts (hunk 1 above)
Comment (burieberry) [reply packages/boxel-ui/addon/src/components/tabbed-header/usage.gts]: making a named block cleaned it up nicely. added a transition, not the same animation, but smoother than before.

Comment (backspace) [reply packages/boxel-ui/addon/src/components/tabbed-header/usage.gts]: nice 🎉

**Pull request merged**